File: templates/lang/android_app_dagger/src_files/test/it/MainActivityInstrumentedTest.java
package {{ .Group }};

import android.app.Instrumentation;
import android.content.Intent;
import android.support.test.InstrumentationRegistry;
import android.support.test.runner.AndroidJUnit4;

import {{ .Group }}.main.MainActivity;

import org.junit.Test;
import org.junit.runner.RunWith;

import static org.assertj.core.api.Assertions.assertThat;


/**
 * Instrumentation test, which will execute on an Android device.
 *
 * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
 */
@RunWith(AndroidJUnit4.class)
public class MainActivityInstrumentedTest {

  @Test
  public void launchMainActivity() throws Exception {
    Instrumentation instrumentation = InstrumentationRegistry.getInstrumentation();
    Instrumentation.ActivityMonitor activityMonitor =
        instrumentation.addMonitor(MainActivity.class.getName(), null, false);


    Intent intent = new Intent(Intent.ACTION_MAIN);
    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
    intent.setClassName(instrumentation.getTargetContext(), MainActivity.class.getName());
    instrumentation.startActivitySync(intent);
    MainActivity activity = (MainActivity) instrumentation.waitForMonitor(activityMonitor);

    assertThat(activity).isNotNull();
    assertThat(activity.getTitle()).isEqualTo("{{ .CamelName }}");
  }
}

File: templates/lang/kotlin_android_app/src_files/MainActivityInstrumentedTest.kt
package {{ .Group }};

import android.content.Intent
import android.support.test.InstrumentationRegistry
import android.support.test.runner.AndroidJUnit4
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith

/**
 * Instrumentation test, which will execute on an Android device.
 *
 * @see [Testing documentation](http://d.android.com/tools/testing)
 */
@RunWith(AndroidJUnit4::class)
class MainActivityInstrumentedTest {

  @Test
  @Throws(Exception::class)
  fun launchMainActivity() {
    val instrumentation = InstrumentationRegistry.getInstrumentation()
    val activityMonitor = instrumentation.addMonitor(MainActivity::class.java.name, null, false)


    val intent = Intent(Intent.ACTION_MAIN)
    intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
    intent.setClassName(instrumentation.targetContext, MainActivity::class.java.name)
    instrumentation.startActivitySync(intent)
    val activity = instrumentation.waitForMonitor(activityMonitor) as MainActivity

    assertThat(activity).isNotNull()
    assertThat(activity.title).isEqualTo("{{ .CamelName }}")
  }
}
